Add `id-length` ESLint rule

ehmicky opened the pull request:
This adds the [`id-length`](https://eslint.org/docs/rules/id-length) ESLint rule.

Even with autocompletion, long variables can be annoying. On the other hand, too short variables can be obscure. So this PR gives a 24 characters maximum per variable, which is hopefully a good balance. Please let me know if this should be decreased or increased.

ehmicky commented on packages/build/src/core/feature_flags.js:
```diff
@@ -12,12 +12,14 @@ const getFeatureFlag = function (name) {
   return { [name]: true }
 }
 
+/* eslint-disable id-length */
```
Feature flags are often long, and that's ok.

ehmicky commented on packages/build/tests/env/fixtures/gatsby_telemetry/plugin.js:
```diff
@@ -1,4 +1,5 @@
 const {
+  // eslint-disable-next-line id-length
```
This is an external environment variable.

erezrokah commented on packages/build/src/core/constants.js:
```diff
@@ -82,12 +82,12 @@ const getEdgeHandlersSrc = async function (edgeHandlers, buildDir) {
 
-  if (await isDirectory(`${buildDir}/${DEFAULT_EDGE_HANDLERS_SRC}`)) {
-    return DEFAULT_EDGE_HANDLERS_SRC
+  if (await isDirectory(`${buildDir}/${DEFAULT_HANDLERS_SRC}`)) {
```
The longer name is more explicit here and I think it is better, WDYT?

ehmicky replied:
:+1:

erezrokah commented on packages/build/src/install/functions.js:
```diff
@@ -2,18 +2,18 @@ const { dirname } = require('path')
 const readdirp = require('readdirp')
 
-const { logInstallFunctionDependencies } = require('../log/messages/install')
+const { logInstallFunctionDeps } = require('../log/messages/install')
```
[sand] I prefer the non abbreviated `logInstallFunctionDependencies` name

ehmicky replied:
:+1:

erezrokah commented on packages/build/src/log/messages/install.js:
```diff
@@ -17,13 +17,13 @@ const logMissingPluginsWarning = function (logs, packages) {
 }
 
-const logInstallLocalPluginsDeps = function (logs, localPluginsOptions) {
+const logInstallLocalDeps = function (logs, localPluginsOptions) {
```
I think the longer name makes sense here, WDYT?

ehmicky replied:
:+1:

erezrokah reviewed (changes requested):
Didn't go over all the changes, since I wanted to start a discussion. 
Min length makes sense, but max length can be too restricting.
I added some comments where the previous version was more explicit and readable (to me).
Please let me know what you think

ehmicky commented:
This makes sense :+1: 
I personally prefer to keep variables not too long, but I agree this might be more of a personal preference, and there are some very valid reasons to keep some variables long.
What would be your preference:
  1. Removing any constraint on the `max` length?
  2. Keeping the `max` length constraint, but with a higher value (in which case, please let me know which would be best)?
Both solutions sound good to me :+1:

erezrokah commented:
> 1. Removing any constraint on the `max` length?

I would go with this one

ehmicky commented:
Sounds good, I removed the `max` constraint.
I also removed all the changes in this PR related to long variable names, so this PR is now much smaller.

ehmicky commented:
Done as part of #2004.
